src/main/java/com/microsoft/azure/relay/ClientWebSocket.java

```diff
@@ -344,6 +341,15 @@ public void onError(Session session, Throwable cause) {
 		RelayLogger.throwingException(cause, this);
 	}
 	
+	private static ByteBuffer deepCopyByteBuffer(ByteBuffer original) {
```

Comment:
>deepCopyByteBuffer [](start = 27, length = 18)

Is deepCopyByteBuffer allowed to mutate the original ByteBuffer by changing the the current of position? What about mark?